service/storage_service.cc

```diff
@@ -377,7 +377,7 @@ future<> storage_service::topology_state_load(cdc::generation_service& cdc_gen_s
                     co_await tmptr->update_normal_tokens(rs.ring.value().tokens, ip);
                 } else {
                     tmptr->add_bootstrap_tokens(rs.ring.value().tokens, ip);
-                    co_await update_pending_ranges(tmptr, format("bootstrapping node {}/{}", id, ip));
+                    co_await update_pending_ranges(tmptr, ::format("bootstrapping node {}/{}", id, ip));
```

Comment:
Maybe the problem is actually that the formatter for tagged_uuid is defined like this:

```
struct fmt::formatter<utils::tagged_uuid<Tag>> : fmt::formatter<std::string_view>
```

Notice that it extends a formatter for string_view. Maybe some fmt impl has a specialization for this formatter, and misinterprets the value as string view? Can you check if avoiding the inheritance would fix the problem?

Reply:
The fact that we see this invoked in the backtrace would support this:
```
fmt::v9::vformat[abi:cxx11](fmt::v9::basic_string_view<char>, fmt::v9::basic_format_args<fmt::v9::basic_format_context<fmt::v9::appender, char> >) at ??:?
```

Reply:
Well, it doesn't compile without the inheritance
```
/usr/include/fmt/core.h:1311:28: error: no member named 'parse' in 'fmt::formatter<utils::UUID>'                                                                                                                                                                                       
    parse_ctx.advance_to(f.parse(parse_ctx));                                                                                              
                         ~ ^                                                                                                               
/usr/include/fmt/core.h:1293:21: note: in instantiation of function template specialization 'fmt::detail::value<fmt::basic_format_context<fmt::appender, char>>::format_custom_arg<utils::UUID, fmt::formatter<utils::UUID>>' requested here
    custom.format = format_custom_arg<                                                                                                     
                    ^                                                                                                                      
/home/kbraun/scylladb/seastar/include/seastar/util/log.hh:226:33: note: in instantiation of function template specialization 'fmt::format_to<seastar::internal::log_buf::inserter_iterator, const utils::UUID &, unsigned long, const unsigned int &, const unsigned int &, 0>' request
ed here                                                                                                                                    
                    return fmt::format_to(it, fmt::runtime(fmt.format), std::forward<Args>(args)...);                                      
                                ^                                                                                                                                                                                                                                                      
/home/kbraun/scylladb/seastar/include/seastar/util/log.hh:385:9: note: in instantiation of function template specialization 'seastar::logger::log<const utils::UUID &, unsigned long, const unsigned int &, const unsigned int &>' requested here
        log(log_level::trace, std::move(fmt), std::forward<Args>(args)...);                                                                
        ^                                                                                                                                  
./tracing/trace_state.hh:136:28: note: in instantiation of function template specialization 'seastar::logger::trace<const utils::UUID &, unsigned long, const unsigned int &, const unsigned int &>' requested here
        trace_state_logger.trace("{}: props {}, slow query threshold {}us, slow query ttl {}s", session_id(), _state_props.mask(), info.slow_query_threshold_us, info.slow_query_ttl_sec);
```

also note that this is private inheritance.

Reply:
You could define parse() instead. But I don't know if inheriting from string_view's formatter is good or not, I saw it in some online examples.

Reply:
I don't see how "misinterpreting the value as string_view" could happen with private inheritance though.

AFAIU it's only used so we can implement `format` conveniently, delegating to `formatter<string_view>::format`.

Reply:
Note that it's a struct, so it's actually a public inheritance.

Reply:
> Note that it's a struct, so it's actually a public inheritance.

Oh.